components/pedigree/resources/src/main/resources/pedigree/view.js

```diff
@@ -717,22 +717,24 @@ define([
 
             //timer.printSinceLast("=== highlight: ");
 
-            // re-evaluate which buttons & handles are appropriate for the nodes (e.g. twin button appears/disappears)
-            for (var nodeID in this._nodeMap) {
-                if (this._nodeMap.hasOwnProperty(nodeID)) {
-                    if (editor.getGraph().isPerson(nodeID) && !this.getNode(nodeID).getGraphics().getHoverBox().isMenuToggled()) {
-                        this.getNode(nodeID).getGraphics().getHoverBox().removeButtons();
-                        this.getNode(nodeID).getGraphics().getHoverBox().removeHandles();
+            if (!editor.isReadOnlyMode()) {
```

Comment:
As far as I understand, I am fixing [PT-2096](https://phenotips.atlassian.net/browse/PT-2096) by totaly disallowing users with read-only rights to ever enter pedigree-editor. Why we still taking care of this here?